packages/flutter/lib/src/painting/text_span.dart

```diff
@@ -353,8 +353,14 @@ class TextSpan extends DiagnosticableTree {
 
   @override
   List<DiagnosticsNode> debugDescribeChildren() {
-    return children == null ?
-        const <DiagnosticsNode>[] :
-        children.map((TextSpan child) => child?.toDiagnosticsNode()).toList();
+    if (children == null)
+      return const <DiagnosticsNode>[];
+    return children.map((TextSpan child) {
+      if (child != null) {
+        return child.toDiagnosticsNode();
+      } else {
+        return new DiagnosticsNode.message('<null child>');
```

Comment:
toStringDeep() is currently handling null children adding <null> when a  a entry from debugDescribeChildren() is null. I'm also fine with making it explicit but if we do we should remove the special case in debugDescribeChildren() that is emitting  '<null>'

Reply:
As far as I can tell toStringDeep doesn't fully handle null children (e.g. it uses `children.first.textTreeConfiguration` without checking for null first). But I do see what you mean that there's some special check in there. I guess I'll remove the check.